crates/core/app/tests/mock_consensus.rs

```diff
@@ -36,3 +37,35 @@ async fn mock_consensus_can_send_an_init_chain_request() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn mock_consensus_can_send_a_single_empty_block() -> anyhow::Result<()> {
+    // Install a test logger, and acquire some temporary storage.
+    let guard = common::set_tracing_subscriber();
+    let storage = TempStorage::new().await?;
+
+    // Instantiate the consensus service, and start the test node.
+    let mut engine = {
+        use penumbra_mock_consensus::TestNode;
+        let app_state = AppState::default();
+        let consensus = Consensus::new(storage.as_ref().clone());
+        TestNode::builder()
+            .single_validator()
+            .with_penumbra_auto_app_state(app_state)?
+            .init_chain(consensus)
+            .await?
+    };
+
+    let block = engine
+        .block()
+        .with_data(vec![])
+        .with_evidence(List::new(Vec::new()))
+        .finish()?;
+    engine.send_block(block).await?;
```

Comment:
in a follow-on, i'd like to explore the idea of letting this builder expose a `send` too. that way one can opt to either `finish` the builder and get a block, or (_on account of the builder already holding a reference to the engine_) immediately send it to the consensus service.

Reply:
Q: what's the use case for _not_ immediately executing the block? why shouldn't `.finish()` just be `.execute()`? would we ever want to have the engine sign a block without executing it?

Reply:
i don't have a concrete use case in mind, it was just marginally easier to implement this way!

i like `.execute()`, i can run with that.

Reply:
f796ff113